packages/fxa-event-broker/lib/notificationProcessor.ts

```diff
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { SQS } from 'aws-sdk';
+import { Logger } from 'mozlog';
+import { Consumer } from 'sqs-consumer';
+import * as joi from 'typesafe-joi';
+
+import { Datastore } from './db';
+
+// Event strings
+const LOGIN_EVENT = 'login';
+const SUBSCRIPTION_UPDATE_EVENT = 'subscription:update';
+
+// Message schemas
+const BASE_MESSAGE_SCHEMA = joi
+  .object()
+  .keys({
+    event: joi.string().required(),
+    uid: joi.string().required()
+  })
+  .unknown(true)
+  .required();
+
+const LOGIN_SCHEMA = joi
+  .object()
+  .keys({
+    clientId: joi.string().required(),
+    deviceCount: joi
```

Comment:
Do we care how strict these schema are? There is a bunch of logic on the auth/content servers for validation that might be able to be shared here. I have a goal of sharing some validation logic in fxa-shared ([@philbooth does too](https://github.com/mozilla/fxa/issues/380)) across the repos.

Reply:
That would definitely be preferred. Ideally the shared one can use the typesafe-joi, as it helpfully adds proper types to the return value based on the schema.

Reply:
I wasn't too picky about how strict they are since I'm not storing them and such at the moment.